core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1052,7 +1057,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   }
 
   private def batchMissingRequiredVerification(batch: MutableRecordBatch, requestVerificationGuard: VerificationGuard): Boolean = {
-    producerStateManager.producerStateManagerConfig().transactionVerificationEnabled() && !batch.isControlBatch &&
+    producerStateManager.producerStateManagerConfig().transactionVerificationEnabled() &&
```

Comment:
What's the reason to move the check for control batch out of this function?

Reply:
I moved the check to outside the if condition (otherwise we can't write commit markers)
I can put this back, but it will just be a duplicate check.

Reply:
No, this is good.  I'm just thinking if having this function makes the code better or worse -- it's really called exactly from one place and looks like the order of checks is important and can lead to subtle bugs.

Reply:
Yeah. I was also wondering if this was the best way.

Reply:
I think the original reason to have the method was to not have too many if conditions on that line. Not sure if it would be useful to have one big method that explains why the checks are ordered in such a way.